keps/sig-storage/625-csi-migration/README.md

```diff
@@ -156,10 +165,155 @@ Running this set of tests is optional for a per-provider basis. We would
 upgrade/downgrade is provider agnostic.
 
+## Production Readiness Review Questionnaire
+
+<!--
+
+Production readiness reviews are intended to ensure that features merging into
+Kubernetes are observable, scalable and supportable; can be safely operated in
+production environments, and can be disabled or rolled back in the event they
+cause increased failures in production. See more in the PRR KEP at
+https://git.k8s.io/enhancements/keps/sig-architecture/1194-prod-readiness.
+
+The production readiness review questionnaire must be completed and approved
+for the KEP to move to `implementable` status and be included in the release.
+
+In some cases, the questions below should also have answers in `kep.yaml`. This
+is to enable automation to verify the presence of the review, and to reduce review
+burden and latency.
+
+The KEP must have a approver from the
+[`prod-readiness-approvers`](http://git.k8s.io/enhancements/OWNERS_ALIASES)
+team. Please reach out on the
+[#prod-readiness](https://kubernetes.slack.com/archives/CPNHUMN74) channel if
+you need any help or guidance.
+
+-->
+
+### Feature Enablement and Rollback
+
+
+* **How can this feature be enabled / disabled in a live cluster?**
+  - [x] Feature gate (also fill in values in `kep.yaml`)
+    - Feature gate name: CSIMigration, CSIMigration{cloud-provider}
+    - Components depending on the feature gate: kubelet, kube-controller-manager
+
+* **Does enabling the feature change any default behavior?**
+  Yes and No. If only CSIMigration feature flag is enabled. Nothing will change on the cluster behavior. However, if CSIMigration && CSIMigration{cloud-provider} are both enabled, the behavior will change. The in-tree volume plugin that the cloud-provider use will be redirect to use the corresponding CSI driver. But from a user perspective, nothing will be noticed.
+
+* **Can the feature be disabled once it has been enabled (i.e. can we roll back
+  the enablement)?**
+  Yes. The `volume.beta.kubernetes.io/migrated-to` annotation will be used by `v1.6.0+` of the CSI External provisioner to determine whether a PV or PVC should be operated on by the provisioner. The annotation will be set (and removed on rollback) for Kubernetes `v1.17.2+`, we will carefully document the fact that rollback with migration on may not be successful to a version before `v1.17.2`. The benefit being that PV Controller start-up annealing of this annotation will allow the PV Controller to stand down and the CSI External Provisioner to pick up a PV that was dynamically provisioned before migration was enabled. These newer external provisioners will still be compatible with older versions of Kubernetes with migration on even if they do not set the migrated-to annotation. However, without the migrated-to annotation a newer provisioner with a Kubernetes cluster `<1.17.2` will not be able to delete volumes provisioned before migration until the Kubernetes cluster is upgraded to `v1.17.2+`.
```

Comment:
1) On which object the annotation is set?
2) Can you clarify in more details what does "rollback with migration on may not be successful" mean?

nit: it would be much easier to comment, if you could split the lines

Reply:
Oops, I was copy-paste from the design doc and it seems not related to rollback with migration off actually. The issue only exists if we rollback to a version before v1.17.2 and the feature flag is still on. For more reading, https://github.com/kubernetes/community/blob/master/contributors/design-proposals/storage/csi-migration.md#persistent-volume-controller